### PR TITLE
Add namespace duplication validation to src/dest ns validate methods

### PR DESCRIPTION
**Pull request by sseago**

### CODE REVIEW

**Comment by djwhatle** [comment]
Here's what this looks like on the API:
image

And here's what it looks like in the UI
image

File: pkg/controller/migplan/validation.go [hunk 1]
```diff
@@ -47,6 +47,8 @@ const (
 	NsLimitExceeded                            = "NamespaceLimitExceeded"
 	NsLengthExceeded                           = "NamespaceLengthExceeded"
 	NsHaveNodeSelectors                        = "NamespacesHaveNodeSelectors"
+	DuplicateNsOnSourceCluster                 = "NamespaceNotFoundOnSourceCluster"
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go]: Is this a copy paste error? The content of the string doesn't seem to match the const name.

File: pkg/controller/migplan/validation.go [hunk 2]
```diff
@@ -683,10 +694,25 @@ func (r ReconcileMigPlan) validateSourceNamespaces(plan *migapi.MigPlan) error {
...
+
+	namespaces := plan.GetDestinationNamespaces()
+	duplicates := listDuplicateNamespaces(namespaces)
+	if len(duplicates) > 0 {
+		plan.Status.SetCondition(migapi.Condition{
+			Type:     DuplicateNsOnDestinationCluster,
+			Status:   True,
+			Reason:   DuplicateNs,
+			Category: Critical,
+			Message:  "Duplicate destination cluster namespaces [] in migplan.",
```
Comment (djwhatle) [inline pkg/controller/migplan/validation.go:709]: could we change "migplan" here to migration plan?

**Review by djwhatle — CHANGES_REQUESTED** [review]

File: pkg/controller/migplan/validation.go (hunk 1 above)
Comment (sseago) [reply pkg/controller/migplan/validation.go]: Yes. I've already fixed it in the commit.

File: pkg/controller/migplan/validation.go (hunk 2 above)
Comment (sseago) [reply pkg/controller/migplan/validation.go:709]: Sure. I was thinking that API-level errors were better using the actual api resource names, but if we're using "migration plan" elsewhere I can do that here.

Comment (sseago) [reply pkg/controller/migplan/validation.go:709]: Ahh. We're inconsistent here. It looks liike we *do* use `migstorage` and `migcluster` in other condition messages, but we also use `migration plan` elsewhere.

Comment (djwhatle) [reply pkg/controller/migplan/validation.go:709]: Yeah, that was my conclusion as well. It could probably go either way to be honest.

**Review by djwhatle — APPROVED** [review]

**Comment by djwhatle** [comment]
@sseago ACK, merge whenever you're ready.

**Pull request merged**